parsing.go

```diff
@@ -131,50 +153,73 @@ func parseMarkdown(f string) ([]*Req, error) {
 					if level == lastHeadingLevel {
 						return nil, fmt.Errorf("requirement heading on line %d at same level as previous heading on line %d (%d): %q", lno, lastHeadingLine, level, line)
 					}
-					start = true
-				} else {
-					// No requirement ID on this heading. Will be ignored.
 				}
 			}
 
-			if end {
-				// Close the current requirement.
-				newReq, err := parseReq(reqBuf.String())
-				if err != nil {
-					return nil, err
-				}
-				reqs = append(reqs, newReq)
-				inReq = false
+			// If we're currently parsing a requirement and it's appropriate close it
+			if (inReq != None) && (headingHasReqID || level < reqLevel) {
+				reqs, err = parseMarkdownFragment(inReq, reqBuf.String(), reqs)
+				inReq = None
 			}
-			if start {
-				// Start a new requirement at the current line.
-				inReq = true
+
+			// If this is the start of a new requirement initialise data
+			if headingHasReqID {
+				inReq = Heading
 				reqLevel = level
 				reqLine = lno
 				reqBuf.Reset()
 				line = title
 			}
+			if level > 0 {
+				lastHeadingLevel = level
+				lastHeadingLine = lno
+			}
+		} else if reTableHeader.MatchString(line) {
+			// It's a requirements table
```

Comment:
so ATX Header requirements can encode sub-requirements, but table requirements cannot?

Reply:
I'm not sure I understand... there's no concept of sub-requirements, but there are additional checks for ATX-style requirements to ensure that the depth of the heading used is consistent for a block of requirements within a section of the document. These rules aren't required for a table of requirements because they are all neatly contained within the table and if a heading is encountered we know it's the end of the table (and hence block of requirements).